gRPC server with full ffmpeg command line

muh200 opened the pull request.

Commit:
Implemented prototype with gRPC server

This has the same functionality as the prototype except that the server
that takes POST requests is replaced with a gRPC server.

Commit:
Made the ffmpeg output stream to the client

Previously, the ffmpeg output would be returned to the client in the
response but this didn't allow the client to see the progress of their
command in real-time.

Commit:
Implemented full ffmpeg command line with gcsfuse.

The server now uses gcsfuse to mount GCS buckets in the local filesystem
which allows ffmpeg to run without any translation of its arguments.

Commit:
Created client CLI to interact with ffmpeg worker.

Commit:
Optimize Dockerfile by leveraging build cache.

By copying the requirements.txt separately, python dependencies do not
have to be installed every time the source code changes.
This results in faster build times.

isasi commented on worker/protos/ffmpeg_worker.proto:
```diff
@@ -0,0 +1,18 @@
+  rpc transcode(Request) returns (stream Log) {}
+}
+
+message Log {
```
It would be good if we return a Response message with a "status" field (which can be an enum and be populated from subprocess exit status ) along with the log.

isasi commented on worker/protos/ffmpeg_worker.proto:
```diff
@@ -0,0 +1,18 @@
+  string text = 1;
+}
+
+message Request {
```
Name them  FfmpegRequest and FfmpegResponse . 
You can follow the conventions described here :
https://cloud.google.com/apis/design/naming_convention?hl=en

isasi commented on worker/run_ffmpeg.py:
```diff
@@ -0,0 +1,85 @@
+# limitations under the License.
+# ==============================================================================
+"""Runs ffmpeg with GCS files by mounting GCS buckets with gcsfuse.
+"""
```
why is this separated into a different file ?

Commit:
Changed naming of request and response messages.

Commit:
Removed the need for demoting privilege.

The purpose of run_ffmpeg.py was to have a child process with lower
privileges than the parent process.
This is not necessary since the parent process could run with lower privileges.
